quinn/src/endpoint.rs

```diff
@@ -318,6 +338,19 @@ impl Endpoint {
     }
 }
 
+// Struct for holding the stats returned from Endpoint::stats
+//
+#[non_exhaustive]
+#[derive(Debug, Default, Copy, Clone)]
+pub struct EndpointStats {
+    pub open_connections: u64,
+    pub incoming_handshakes: u64,
```

Comment:
If `incoming_handshakes` is merely the sum of accepted, refused and ignored handshakes I think we should track those three numbers separately without additionally tracking `incoming_handshake`, since it can trivially derived from the other three (that is, replace this with `accepted_handshakes`).